src/Language/Java/Inline/Plugin.hs

```diff
@@ -83,9 +85,12 @@ plugin = defaultPlugin
           return guts
             { mg_binds = binds
             , mg_foreign = appendStubC (mg_foreign guts) $
-                             text bctable_header
+                        CStub
+                          (  text bctable_header
                            $$ dotClasses dcs
                            $$ cConstructors
+                          )
+                          [] []
```

Comment:
This is the most uncertain part of the change. It seems to work, but the documentation of `CStub` doesn't promise any support for constructor functions.